src/Mono.Android/Android.Runtime/JNIEnvInit.cs

```diff
@@ -48,6 +49,8 @@ internal struct JnienvInitializeArgs {
 
 		internal static JniRuntime? androidRuntime;
 
+		public static DotNetRuntimeType RuntimeType { get; private set; } = DotNetRuntimeType.Unknown;
```

Comment:
Could this value be set by a trimmer switch instead of passing it in?

* https://github.com/dotnet/runtime/blob/main/docs/tools/illink/data-formats.md#substitute-method-body-with-a-constant

The build knows the runtime it targets, so it could do the work at build time.

This would make this property treated as a constant and code for other runtimes would be trimmed away.

Reply:
I think that eventually turning it into a trimmer feature is the goal. But right now, I need it to be a runtime check since I sometimes work without the trimmer

Reply:
Trimmer feature switches use `AppContext.TryGetSwitch()` if the app isn't trimmed, here is an example:

* https://github.com/dotnet/android/pull/9811/files#diff-1a79098cae925c797ae39c6cc09768aa3812291002ba44a341121fa0c2e0f064

Reply:
That might be the better solution, but we can make that change at a later time, when everything is in `main`, I think